src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -419,6 +405,11 @@ public Path getJavaAppEngineSdkPath() {
     return getSdkPath().resolve(JAVA_APPENGINE_SDK_PATH);
   }
 
+  @VisibleForTesting
+  Path getJavaBinaryPath() {
+    return javaHomePath.resolve("bin/java");
```

Comment:
Perhaps this should be `getJavaExecutable()`. It should do an is-windows test and check for the java.exe.